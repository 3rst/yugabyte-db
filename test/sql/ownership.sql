\unset ECHO
\i test/setup.sql

SELECT plan(12);
--SELECT * FROM no_plan();

/****************************************************************************/
SELECT * From check_test(
    db_owner_is(current_database(), current_user, 'mumble'),
	true,
    'db_owner_is(db, user, desc)',
    'mumble',
    ''
);

SELECT * From check_test(
    db_owner_is(current_database(), current_user),
	true,
    'db_owner_is(db, user)',
    'Database ' || quote_ident(current_database()) || ' should be owned by ' || current_user,
    ''
);

SELECT * From check_test(
    db_owner_is('__not__' || current_database(), current_user, 'mumble'),
	false,
    'db_owner_is(non-db, user)',
    'mumble',
    '    Database __not__' || current_database() || ' does not exist'
);

SELECT * From check_test(
    db_owner_is(current_database(), '__not__' || current_user, 'mumble'),
	false,
    'db_owner_is(db, non-user)',
    'mumble',
    '        have: ' || current_user || '
        want: __not__' || current_user
);

/****************************************************************************/
-- Finish the tests and clean up.
SELECT * FROM finish();
ROLLBACK;

