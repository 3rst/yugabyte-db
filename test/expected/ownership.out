\unset ECHO
1..81
ok 1 - db_owner_is(db, user, desc) should pass
ok 2 - db_owner_is(db, user, desc) should have the proper description
ok 3 - db_owner_is(db, user, desc) should have the proper diagnostics
ok 4 - db_owner_is(db, user) should pass
ok 5 - db_owner_is(db, user) should have the proper description
ok 6 - db_owner_is(db, user) should have the proper diagnostics
ok 7 - db_owner_is(non-db, user) should fail
ok 8 - db_owner_is(non-db, user) should have the proper description
ok 9 - db_owner_is(non-db, user) should have the proper diagnostics
ok 10 - db_owner_is(db, non-user) should fail
ok 11 - db_owner_is(db, non-user) should have the proper description
ok 12 - db_owner_is(db, non-user) should have the proper diagnostics
ok 13 - rel_owner_is(sch, tab, user, desc) should pass
ok 14 - rel_owner_is(sch, tab, user, desc) should have the proper description
ok 15 - rel_owner_is(sch, tab, user, desc) should have the proper diagnostics
ok 16 - rel_owner_is(sch, tab, user) should pass
ok 17 - rel_owner_is(sch, tab, user) should have the proper description
ok 18 - rel_owner_is(sch, tab, user) should have the proper diagnostics
ok 19 - rel_owner_is(non-sch, tab, user) should fail
ok 20 - rel_owner_is(non-sch, tab, user) should have the proper description
ok 21 - rel_owner_is(non-sch, tab, user) should have the proper diagnostics
ok 22 - rel_owner_is(sch, non-tab, user) should fail
ok 23 - rel_owner_is(sch, non-tab, user) should have the proper description
ok 24 - rel_owner_is(sch, non-tab, user) should have the proper diagnostics
ok 25 - rel_owner_is(tab, user, desc) should pass
ok 26 - rel_owner_is(tab, user, desc) should have the proper description
ok 27 - rel_owner_is(tab, user, desc) should have the proper diagnostics
ok 28 - rel_owner_is(tab, user) should pass
ok 29 - rel_owner_is(tab, user) should have the proper description
ok 30 - rel_owner_is(tab, user) should have the proper diagnostics
ok 31 - rel_owner_is(non-tab, user) should fail
ok 32 - rel_owner_is(non-tab, user) should have the proper description
ok 33 - rel_owner_is(non-tab, user) should have the proper diagnostics
ok 34 - rel_owner_is(sch, seq, user, desc) should pass
ok 35 - rel_owner_is(sch, seq, user, desc) should have the proper description
ok 36 - rel_owner_is(sch, seq, user, desc) should have the proper diagnostics
ok 37 - rel_owner_is(sch, seq, user) should pass
ok 38 - rel_owner_is(sch, seq, user) should have the proper description
ok 39 - rel_owner_is(sch, seq, user) should have the proper diagnostics
ok 40 - rel_owner_is(non-sch, seq, user) should fail
ok 41 - rel_owner_is(non-sch, seq, user) should have the proper description
ok 42 - rel_owner_is(non-sch, seq, user) should have the proper diagnostics
ok 43 - rel_owner_is(sch, non-seq, user) should fail
ok 44 - rel_owner_is(sch, non-seq, user) should have the proper description
ok 45 - rel_owner_is(sch, non-seq, user) should have the proper diagnostics
ok 46 - rel_owner_is(seq, user, desc) should pass
ok 47 - rel_owner_is(seq, user, desc) should have the proper description
ok 48 - rel_owner_is(seq, user, desc) should have the proper diagnostics
ok 49 - rel_owner_is(seq, user) should pass
ok 50 - rel_owner_is(seq, user) should have the proper description
ok 51 - rel_owner_is(seq, user) should have the proper diagnostics
ok 52 - rel_owner_is(non-seq, user) should fail
ok 53 - rel_owner_is(non-seq, user) should have the proper description
ok 54 - rel_owner_is(non-seq, user) should have the proper diagnostics
ok 55 - table_owner_is(sch, tab, user, desc) should pass
ok 56 - table_owner_is(sch, tab, user, desc) should have the proper description
ok 57 - table_owner_is(sch, tab, user, desc) should have the proper diagnostics
ok 58 - table_owner_is(sch, tab, user) should pass
ok 59 - table_owner_is(sch, tab, user) should have the proper description
ok 60 - table_owner_is(sch, tab, user) should have the proper diagnostics
ok 61 - table_owner_is(non-sch, tab, user) should fail
ok 62 - table_owner_is(non-sch, tab, user) should have the proper description
ok 63 - table_owner_is(non-sch, tab, user) should have the proper diagnostics
ok 64 - table_owner_is(sch, non-tab, user) should fail
ok 65 - table_owner_is(sch, non-tab, user) should have the proper description
ok 66 - table_owner_is(sch, non-tab, user) should have the proper diagnostics
ok 67 - table_owner_is(tab, user, desc) should pass
ok 68 - table_owner_is(tab, user, desc) should have the proper description
ok 69 - table_owner_is(tab, user, desc) should have the proper diagnostics
ok 70 - table_owner_is(tab, user) should pass
ok 71 - table_owner_is(tab, user) should have the proper description
ok 72 - table_owner_is(tab, user) should have the proper diagnostics
ok 73 - table_owner_is(non-tab, user) should fail
ok 74 - table_owner_is(non-tab, user) should have the proper description
ok 75 - table_owner_is(non-tab, user) should have the proper diagnostics
ok 76 - table_owner_is(sch, seq, user, desc) should fail
ok 77 - table_owner_is(sch, seq, user, desc) should have the proper description
ok 78 - table_owner_is(sch, seq, user, desc) should have the proper diagnostics
ok 79 - table_owner_is(seq, user, desc) should fail
ok 80 - table_owner_is(seq, user, desc) should have the proper description
ok 81 - table_owner_is(seq, user, desc) should have the proper diagnostics
